api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -206,6 +204,17 @@ message ExternalProcessor {
   // Instead, the stream to the external processor will be closed. There will be no
   // more external processing for this stream from now on.
   bool disable_immediate_response = 15;
+
+  // Specifies a list of metadata namespaces whose values, if present, will be passed to the
+  // ext_proc service as an opaque *protobuf::Struct*.
+  repeated string metadata_context_namespaces = 16;
+
+  // Specifies a list of metadata namespaces whose values, if present, will be passed to the
+  // ext_proc service. :ref:`typed_filter_metadata <envoy_v3_api_field_config.core.v3.Metadata.typed_filter_metadata>` is passed as an ``protobuf::Any``.
+  //
+  // It works in a way similar to ``metadata_context_namespaces`` but allows envoy and external processing server to share the protobuf message definition
+  // in order to do a safe parsing.
+  repeated string typed_metadata_context_namespaces = 17;
```

Comment:
Could we please add a knob for disabling this on Envoy side? just like disable immediate response etc..

Given the growing list of knobs, I wonder if this is an opportunity to consolidate all these enable/disable toggles into a message called serverCapabilities that has header mutation rules, allow_immediate_response, allow_dynamic_metadata, mode overrides, etc. And this capability proto should technically be advertised in the ProcessingRequest as well so that the ext_proc server knows what is and what is not allowed by the server.

Even if there is no appetite to do this capability thingy, we really need a boolean guard for every new piece of data that we will allow Envoy to accept from the Ext proc server. Some of our systems use this filter to talk to untrusted services and we would like to be strict about what things our data plane ingests from a remote ext_proc server. Hence the reason for toggles like immediate response, mode overrides, etc.

Reply:
also, naming suggestion: accepted_typed_metadata_namespaces

Reply:
I've updated the proto changes to enable route-level overrides and in the process cleaned up this naming a bit.